build.xml

```diff
@@ -22,18 +23,339 @@
 		<pathelement location="${env.CATALINA_HOME}/lib/catalina-ant.jar"/>
 	</path>
 
-	<target name="clean_and_tomcat_deploy" description="Deploy prepared WAR file from project dir to Tomcat webapps folder">
-		<delete file="${env.CATALINA_HOME}/webapps/ScadaBR.war"></delete>
-		<delete dir="${env.CATALINA_HOME}/webapps/ScadaBR"></delete>
-		<copy file="ScadaBR.war" todir="${env.CATALINA_HOME}/webapps" overwrite="true">
+	<target name="init-core" description="Prepare project before build">
+		<delete dir="${build-project.dir}"/>
+		<mkdir dir="${build-project.dir}"/>
+		<mkdir dir="${build-project.src.dir}"/>
+		<mkdir dir="${build-project.test.dir}"/>
+		<mkdir dir="${build-project.log.dir}"/>
+		<copy includeemptydirs="false" todir="${build-project.src.dir}">
+			<fileset dir="src">
+				<exclude name="**/*.launch"/>
+				<exclude name="**/*.xtend"/>
+				<exclude name="**/*.java"/>
+			</fileset>
+		</copy>
+		<copy includeemptydirs="false" todir="${build-project.test.dir}">
+			<fileset dir="test">
+				<exclude name="**/*.launch"/>
+				<exclude name="**/*.xtend"/>
+				<exclude name="**/*.java"/>
+			</fileset>
 		</copy>
 	</target>
 
-	<target name="tomcat_clean_static_content" description="Delete static content">
-		<delete dir="${env.CATALINA_HOME}/work/Catalina/localhost"></delete>
+	<target name="build-core" description="Build Scada-LTS core application">
```

Comment:
`build-core` is a basic task that is a part of the `build` tasks group.  
Basic task should not depends on any task. To perform 'build' with just a java code use `"build-no-ui-skip-test"`
```
<target name="build" description="Perform project build workflow" depends="
	init-core,
	init-static-dependencies,
	init-ui-dependencies,
	build-core,
	test-junit,
	test-frontend-unit,
	build-ui
"/>
```

Reply:
Yes, I know you took this approach, but here I propose a small compromise that will allow the target to fire autonomously. It is worth minimizing the dependencies, but also trying to find a compromise so that as many targets are independent as possible.